agr_java_core/src/main/java/org/alliancegenome/neo4j/entity/node/Allele.java

```diff
@@ -112,5 +112,11 @@ public void setCategory(String category) {
         // this is a calculated property but the setter needs to be here
         // for deserialization purposes.
     }
+    public void addVariant(Variant entity) {
+        if (variants == null)
+            variants = new ArrayList<>();
+        if (!variants.contains(entity))
```

Comment:
WE should consider making variants a Set rather than a List to inherently have non-duplicate values.